oauth.go

```diff
@@ -143,19 +148,91 @@ func GetKeyvaultToken(grantType OAuthGrantType, configFilePath string) (authoriz
 			return nil, err
 		}
 
-		servicePrincipalToken, err := GetServicePrincipalToken(&config.AzureAuthConfig, env)
+		servicePrincipalToken, err := GetServicePrincipalToken(&config.AzureAuthConfig, env, resource)
 		if err != nil {
 			return nil, err
 		}
 		authorizer = autorest.NewBearerAuthorizer(servicePrincipalToken)
 		return authorizer, nil
-	} else {
-		return
+	} 
+	return
+}
+func GetSubscriptionId(configFilePath string) (subscriptionId *string, err error) {
```

Comment:
isn't subscription id part of `azure.json`? just carry `AzureConfig` as part of your data structure. (same for the rest of the fields)